x/wasm/keeper/api.go

```diff
@@ -1,129 +1,43 @@
 }
 
-// returns result, gas used, error
-func (a cosmwasmAPIImpl) callCallablePoint(contractAddrStr string, name []byte, args []byte, isReadonly bool, callstack []byte, gasLimit uint64) ([]byte, uint64, error) {
-	contractAddr := sdk.MustAccAddressFromBech32(contractAddrStr)
-	contractInfo, codeInfo, prefixStore, err := a.keeper.contractInstance(*a.ctx, contractAddr)
-	if err != nil {
-		return nil, 0, err
-	}
-
-	if a.keeper.IsInactiveContract(*a.ctx, contractAddr) {
-		return nil, 0, fmt.Errorf("called contract cannot be executed")
-	}
-
-	env := types.NewEnv(*a.ctx, contractAddr)
-	wasmStore := types.NewWasmStore(prefixStore)
-	gasMultiplier := a.keeper.getGasMultiplier(*a.ctx)
-	querier := NewQueryHandler(*a.ctx, a.keeper.wasmVMQueryHandler, contractAddr, gasMultiplier)
-	gasMeter := a.keeper.gasMeter(*a.ctx)
-	api := a.keeper.cosmwasmAPI(*a.ctx)
-
-	instantiateCost := gasMultiplier.ToWasmVMGas(a.keeper.instantiateContractCosts(a.keeper.gasRegister, *a.ctx, a.keeper.IsPinnedCode(*a.ctx, contractInfo.CodeID), len(args)))
-	if gasLimit < instantiateCost {
-		return nil, 0, fmt.Errorf("lack of gas for calling callable point")
-	}
-	wasmGasLimit := gasLimit - instantiateCost
-
-	result, events, attrs, gas, err := a.keeper.wasmVM.CallCallablePoint(name, codeInfo.CodeHash, isReadonly, callstack, env, args, wasmStore, api, querier, gasMeter, wasmGasLimit, costJSONDeserialization)
-	gas += instantiateCost
-
-	if err != nil {
-		return nil, gas, err
-	}
-
-	if !isReadonly {
-		// issue events and attrs
-		if len(attrs) != 0 {
-			eventsByAttr, err := newCallablePointEvent(attrs, contractAddr, callstack)
-			if err != nil {
-				return nil, gas, err
-			}
-			a.ctx.EventManager().EmitEvents(eventsByAttr)
-		}
-
-		if len(events) != 0 {
-			customEvents, err := newCustomCallablePointEvents(events, contractAddr, callstack)
-			if err != nil {
-				return nil, gas, err
-			}
-			a.ctx.EventManager().EmitEvents(customEvents)
-		}
-	}
-
-	return result, gas, err
-}
-
-// returns result, gas used, error
-func (a cosmwasmAPIImpl) validateInterface(contractAddrStr string, expectedInterface []byte) ([]byte, uint64, error) {
-	contractAddr := sdk.MustAccAddressFromBech32(contractAddrStr)
-
-	if a.keeper.IsInactiveContract(*a.ctx, contractAddr) {
-		return nil, 0, fmt.Errorf("try to validate a contract cannot be executed")
-	}
-
-	_, codeInfo, _, err := a.keeper.contractInstance(*a.ctx, contractAddr)
-	if err != nil {
-		return nil, 0, err
-	}
-
-	result, err := a.keeper.wasmVM.ValidateDynamicLinkInterface(codeInfo.CodeHash, expectedInterface)
-
-	return result, 0, err
-}
-
-func (k Keeper) cosmwasmAPI(ctx sdk.Context) wasmvm.GoAPI {
-	x := cosmwasmAPIImpl{
-		keeper: &k,
-		ctx:    &ctx,
-	}
-	return wasmvm.GoAPI{
-		HumanAddress:      x.humanAddress,
-		CanonicalAddress:  x.canonicalAddress,
-		CallCallablePoint: x.callCallablePoint,
-		ValidateInterface: x.validateInterface,
-	}
+var cosmwasmAPI = wasmvm.GoAPI{ // nolint:varcheck,deadcode
```

Comment:
It is being used in the following format `k.pluskeeper.CosmwasmAPI(ctx)`. If the `var cosmwasmAPI` is no longer in use, it may be a good idea to delete it instead of nolinting it.

Reply:
I agree with you, but I thought it would be better to make it similar to original as much as possible. 
That's why I added `nolint:varcheck,deadcode`.